### ISSUE TITLE
[JENKINS-16979] "Discard old builds" broken in 1.503

### ISSUE THREAD

**Issue by jenkins-infra-bot** [jenkinsci/jenkins#14347]
<p>I just upgraded from Jenkins 1.502 to 1.503. "Discard old builds" has been broken.</p>

<p>When I go to the "Manage Jenkins" page, I see "You have data stored in an older format and/or unreadable data.", and clicking on "Manage" shows "InstantiationError: jenkins.model.BuildDiscarder" for most of my jobs. All my jobs that previously had "discard old builds" specified don’t have it ticked any more.</p>

<p>I presume this is related to this changelog entry: "Discard old build records" behavior is now pluggable, allowing plugins to define custom logic.</p>

<p>If I edit a job and tick "Discard Old Builds", previous values are not there, and there's some spurious text on the page "//x", presumably something that was intended to be removed when this was released.</p>

---
<details><summary><i>Originally reported by <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=mwebber">mwebber</a>, imported from: <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-16979" target="_blank">"Discard old builds" broken in 1.503</a></i></summary>
<i><ul>
<li><b>assignee</b>: <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/kohsuke.png" title="kohsuke's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=kohsuke">kohsuke</a>
<li><b>status</b>: Resolved
<li><b>priority</b>: Blocker
<li><b>component(s)</b>: core
<li><b>resolution</b>: Fixed
<li><b>resolved</b>: 2013-03-19T19:50:06+00:00
<li><b>votes</b>: 12
<li><b>watchers</b>: 20
<li><b>imported</b>: 2025-11-24
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p>I just upgraded from Jenkins 1.502 to 1.503. "Discard old builds" has been broken.</p>

<p>When I go to the "Manage Jenkins" page, I see "You have data stored in an older format and/or unreadable data.", and clicking on "Manage" shows "InstantiationError: jenkins.model.BuildDiscarder" for most of my jobs. All my jobs that previously had "discard old builds" specified don’t have it ticked any more.</p>

<p>I presume this is related to this changelog entry: "Discard old build records" behavior is now pluggable, allowing plugins to define custom logic.</p>

<p>If I edit a job and tick "Discard Old Builds", previous values are not there, and there's some spurious text on the page "//x", presumably something that was intended to be removed when this was released.</p></pre>
</details>
</details>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/kutzi.jpg" title="kutzi's avatar" /> kutzi:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-16979?focusedId=174613&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-174613">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>See commit <a href="https://github.com/jenkinsci/jenkins/commit/f3a2ae31d50d6be6f053a4083b66f3f955fa6248" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/jenkins/commit/f3a2ae31d50d6be6f053a4083b66f3f955fa6248</a></p>
</pre>
</details>

</ul>
</details>

<p>See commit <a href="https://github.com/jenkinsci/jenkins/commit/f3a2ae31d50d6be6f053a4083b66f3f955fa6248" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/jenkins/commit/f3a2ae31d50d6be6f053a4083b66f3f955fa6248</a></p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>asloan7:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-16979?focusedId=174634&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-174634">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Was this change coordinated with the Configuration Slicer? 
It has Discard Old Builds Slicer, Discard Old Builds Slicer, Discard Old Builds Slicer, Discard Old Builds Slicer which I take it are now obsolete... and would be nice if the Configuration Slicer could accommodate the new way.</p>
</pre>
</details>

</ul>
</details>

<p>Was this change coordinated with the Configuration Slicer? <br/>
It has Discard Old Builds Slicer, Discard Old Builds Slicer, Discard Old Builds Slicer, Discard Old Builds Slicer which I take it are now obsolete... and would be nice if the Configuration Slicer could accommodate the new way.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/alexlehm.png" title="alexlehm's avatar" /> alexlehm:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-16979?focusedId=174704&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-174704">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>I noticed the same problem, I assume the old configuration class doesn't migrate the settings to the new one leaving the settings empty.</p>

</pre>
</details>

</ul>
</details>

<p>I noticed the same problem, I assume the old configuration class doesn't migrate the settings to the new one leaving the settings empty.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>asloan7:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-16979?focusedId=174707&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-174707">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>and now I have 250+ build configurations to fix</p>
</pre>
</details>

</ul>
</details>

<p>and now I have 250+ build configurations to fix</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>mwebber:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-16979?focusedId=174713&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-174713">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>&gt;&gt; and now I have 250+ build configurations to fix
If you roll back to 1.502 then all your configurations will be as previosuly, assuming that you didn't "discard old data".</p>
</pre>
</details>

</ul>
</details>

<p>>> and now I have 250+ build configurations to fix<br/>
If you roll back to 1.502 then all your configurations will be as previosuly, assuming that you didn't "discard old data".</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>martouf:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-16979?focusedId=174721&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-174721">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>I believe introduction of this commit results in a number of issues with configuration slicer the ability to set discard old builds and the plugin also creates a build failure error below</p>

<p>12:40:55 ERROR: Publisher org.jenkinsci.plugins.discardbuild.DiscardBuildPublisher aborted due to exception
12:40:55 java.lang.UnsupportedOperationException
12:40:55 	at java.util.AbstractList.add(Unknown Source)
12:40:55 	at java.util.AbstractList.add(Unknown Source)
12:40:55 	at org.jenkinsci.plugins.discardbuild.DiscardBuildPublisher.perform(DiscardBuildPublisher.java:148)
12:40:55 	at hudson.tasks.BuildStepMonitor$3.perform(BuildStepMonitor.java:36)
12:40:55 	at hudson.model.AbstractBuild$AbstractBuildExecution.perform(AbstractBuild.java:814)
12:40:55 	at hudson.model.AbstractBuild$AbstractBuildExecution.performAllBuildSteps(AbstractBuild.java:786)
12:40:55 	at hudson.model.Build$BuildExecution.post2(Build.java:183)
12:40:55 	at hudson.model.AbstractBuild$AbstractBuildExecution.post(AbstractBuild.java:733)
12:40:55 	at hudson.model.Run.execute(Run.java:1592)
12:40:55 	at hudson.model.FreeStyleBuild.run(FreeStyleBuild.java:46)
12:40:55 	at hudson.model.ResourceController.execute(ResourceController.java:88)
12:40:55 	at hudson.model.Executor.run(Executor.java:237)</p>

<p>Makes this version of Jenkins unusable for me, issue needs fixing or capability removing until fixed.</p>
</pre>
</details>

</ul>
</details>

<p>I believe introduction of this commit results in a number of issues with configuration slicer the ability to set discard old builds and the plugin also creates a build failure error below</p>

<p>12:40:55 ERROR: Publisher org.jenkinsci.plugins.discardbuild.DiscardBuildPublisher aborted due to exception<br/>
12:40:55 java.lang.UnsupportedOperationException<br/>
12:40:55 	at java.util.AbstractList.add(Unknown Source)<br/>
12:40:55 	at java.util.AbstractList.add(Unknown Source)<br/>
12:40:55 	at org.jenkinsci.plugins.discardbuild.DiscardBuildPublisher.perform(DiscardBuildPublisher.java:148)<br/>
12:40:55 	at hudson.tasks.BuildStepMonitor$3.perform(BuildStepMonitor.java:36)<br/>
12:40:55 	at hudson.model.AbstractBuild$AbstractBuildExecution.perform(AbstractBuild.java:814)<br/>
12:40:55 	at hudson.model.AbstractBuild$AbstractBuildExecution.performAllBuildSteps(AbstractBuild.java:786)<br/>
12:40:55 	at hudson.model.Build$BuildExecution.post2(Build.java:183)<br/>
12:40:55 	at hudson.model.AbstractBuild$AbstractBuildExecution.post(AbstractBuild.java:733)<br/>
12:40:55 	at hudson.model.Run.execute(Run.java:1592)<br/>
12:40:55 	at hudson.model.FreeStyleBuild.run(FreeStyleBuild.java:46)<br/>
12:40:55 	at hudson.model.ResourceController.execute(ResourceController.java:88)<br/>
12:40:55 	at hudson.model.Executor.run(Executor.java:237)</p>

<p>Makes this version of Jenkins unusable for me, issue needs fixing or capability removing until fixed.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/kohsuke.png" title="kohsuke's avatar" /> kohsuke:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-16979?focusedId=174736&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-174736">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Oh boy, what have I done.</p>
</pre>
</details>

</ul>
</details>

<p>Oh boy, what have I done.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/scm_issue_link.png" title="scm_issue_link's avatar" /> scm_issue_link:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-16979?focusedId=174738&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-174738">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Code changed in jenkins
User: Kohsuke Kawaguchi
Path:
 changelog.html
 core/src/main/java/jenkins/model/BuildDiscarder.java
 test/src/test/groovy/jenkins/model/BuildDiscarderTest.groovy
 test/src/test/resources/jenkins/model/BuildDiscarderTest/testCompatibility/config.xml
 test/src/test/resources/jenkins/model/BuildDiscarderTest/testCompatibility/jobs/foo/config.xml
<a href="http://jenkins-ci.org/commit/jenkins/a33e147a0d8b48d9da17d8dee6a9debbceab8323" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins-ci.org/commit/jenkins/a33e147a0d8b48d9da17d8dee6a9debbceab8323</a>
Log:
  <span class="error">&#91;FIXED JENKINS-16979&#93;</span></p>

<p>Added backward compatible data loader and a corresponding test case.</p>


</pre>
</details>

</ul>
</details>

<p>Code changed in jenkins<br/>
User: Kohsuke Kawaguchi<br/>
Path:<br/>
 changelog.html<br/>
 core/src/main/java/jenkins/model/BuildDiscarder.java<br/>
 test/src/test/groovy/jenkins/model/BuildDiscarderTest.groovy<br/>
 test/src/test/resources/jenkins/model/BuildDiscarderTest/testCompatibility/config.xml<br/>
 test/src/test/resources/jenkins/model/BuildDiscarderTest/testCompatibility/jobs/foo/config.xml<br/>
<a href="http://jenkins-ci.org/commit/jenkins/a33e147a0d8b48d9da17d8dee6a9debbceab8323" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins-ci.org/commit/jenkins/a33e147a0d8b48d9da17d8dee6a9debbceab8323</a><br/>
Log:<br/>
  <span class="error">&#91;FIXED JENKINS-16979&#93;</span></p>

<p>Added backward compatible data loader and a corresponding test case.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/dogfood.png" title="dogfood's avatar" /> dogfood:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-16979?focusedId=174740&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-174740">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Integrated in <span class="image-wrap" style=""><img src="http://ci.jenkins-ci.org/images/16x16/yellow.png" style="border: 0px solid black" /></span> <a href="http://ci.jenkins-ci.org/job/jenkins_main_trunk/2321/" class="external-link" target="_blank" rel="nofollow noopener">jenkins_main_trunk #2321</a>
     <span class="error">&#91;FIXED JENKINS-16979&#93;</span> (Revision a33e147a0d8b48d9da17d8dee6a9debbceab8323)</p>

<p>     Result = UNSTABLE
kohsuke : <a href="https://github.com/jenkinsci/jenkins/commit/a33e147a0d8b48d9da17d8dee6a9debbceab8323" class="external-link" target="_blank" rel="nofollow noopener">a33e147a0d8b48d9da17d8dee6a9debbceab8323</a>
Files : </p>
<ul>
	<li>core/src/main/java/jenkins/model/BuildDiscarder.java</li>
	<li>changelog.html</li>
	<li>test/src/test/resources/jenkins/model/BuildDiscarderTest/testCompatibility/jobs/foo/config.xml</li>
	<li>test/src/test/groovy/jenkins/model/BuildDiscarderTest.groovy</li>
	<li>test/src/test/resources/jenkins/model/BuildDiscarderTest/testCompatibility/config.xml</li>
</ul>

</pre>
</details>

</ul>
</details>

<p>Integrated in <span class="image-wrap" style=""><img src="http://ci.jenkins-ci.org/images/16x16/yellow.png" style="border: 0px solid black" /></span> <a href="http://ci.jenkins-ci.org/job/jenkins_main_trunk/2321/" class="external-link" target="_blank" rel="nofollow noopener">jenkins_main_trunk #2321</a><br/>
     <span class="error">&#91;FIXED JENKINS-16979&#93;</span> (Revision a33e147a0d8b48d9da17d8dee6a9debbceab8323)</p>

<p>     Result = UNSTABLE<br/>
kohsuke : <a href="https://github.com/jenkinsci/jenkins/commit/a33e147a0d8b48d9da17d8dee6a9debbceab8323" class="external-link" target="_blank" rel="nofollow noopener">a33e147a0d8b48d9da17d8dee6a9debbceab8323</a><br/>
Files : </p>
<ul>
	<li>core/src/main/java/jenkins/model/BuildDiscarder.java</li>
	<li>changelog.html</li>
	<li>test/src/test/resources/jenkins/model/BuildDiscarderTest/testCompatibility/jobs/foo/config.xml</li>
	<li>test/src/test/groovy/jenkins/model/BuildDiscarderTest.groovy</li>
	<li>test/src/test/resources/jenkins/model/BuildDiscarderTest/testCompatibility/config.xml</li>
</ul>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>msolnit:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-16979?focusedId=174827&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-174827">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>If I re-apply the settings in build 1.503, will they work correctly?</p>
</pre>
</details>

</ul>
</details>

<p>If I re-apply the settings in build 1.503, will they work correctly?</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/hx_unbanned.png" title="hx_unbanned's avatar" /> hx_unbanned:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-16979?focusedId=174830&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-174830">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>As long as there are no unit-test or test-case, I would say NO. Do not count on that... note that only after creating test for Build queue and the issue of missing build jesse G. found some interesting bugs ... I wonder why this is not tested too ...</p>
</pre>
</details>

</ul>
</details>

<p>As long as there are no unit-test or test-case, I would say NO. Do not count on that... note that only after creating test for Build queue and the issue of missing build jesse G. found some interesting bugs ... I wonder why this is not tested too ...</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/bedge.jpg" title="bedge's avatar" /> bedge:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-16979?focusedId=175631&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-175631">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Looks like it's broken again in 1.506</p>

<p>Discard old builds...
ERROR: Publisher org.jenkinsci.plugins.discardbuild.DiscardBuildPublisher aborted due to exception
java.lang.UnsupportedOperationException
	at java.util.AbstractList.add(AbstractList.java:148)
	at java.util.AbstractList.add(AbstractList.java:108)
	at org.jenkinsci.plugins.discardbuild.DiscardBuildPublisher.perform(DiscardBuildPublisher.java:148)
	at hudson.tasks.BuildStepMonitor$3.perform(BuildStepMonitor.java:36)
	at hudson.model.AbstractBuild$AbstractBuildExecution.perform(AbstractBuild.java:820)
	at hudson.model.AbstractBuild$AbstractBuildExecution.performAllBuildSteps(AbstractBuild.java:792)
	at hudson.model.Build$BuildExecution.post2(Build.java:183)
	at hudson.model.AbstractBuild$AbstractBuildExecution.post(AbstractBuild.java:739)
	at hudson.model.Run.execute(Run.java:1592)
	at hudson.model.FreeStyleBuild.run(FreeStyleBuild.java:46)
	at hudson.model.ResourceController.execute(ResourceController.java:88)
	at hudson.model.Executor.run(Executor.java:237)</p>
</pre>
</details>

</ul>
</details>

<p>Looks like it's broken again in 1.506</p>

<p>Discard old builds...<br/>
ERROR: Publisher org.jenkinsci.plugins.discardbuild.DiscardBuildPublisher aborted due to exception<br/>
java.lang.UnsupportedOperationException<br/>
	at java.util.AbstractList.add(AbstractList.java:148)<br/>
	at java.util.AbstractList.add(AbstractList.java:108)<br/>
	at org.jenkinsci.plugins.discardbuild.DiscardBuildPublisher.perform(DiscardBuildPublisher.java:148)<br/>
	at hudson.tasks.BuildStepMonitor$3.perform(BuildStepMonitor.java:36)<br/>
	at hudson.model.AbstractBuild$AbstractBuildExecution.perform(AbstractBuild.java:820)<br/>
	at hudson.model.AbstractBuild$AbstractBuildExecution.performAllBuildSteps(AbstractBuild.java:792)<br/>
	at hudson.model.Build$BuildExecution.post2(Build.java:183)<br/>
	at hudson.model.AbstractBuild$AbstractBuildExecution.post(AbstractBuild.java:739)<br/>
	at hudson.model.Run.execute(Run.java:1592)<br/>
	at hudson.model.FreeStyleBuild.run(FreeStyleBuild.java:46)<br/>
	at hudson.model.ResourceController.execute(ResourceController.java:88)<br/>
	at hudson.model.Executor.run(Executor.java:237)</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/bedge.jpg" title="bedge's avatar" /> bedge:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-16979?focusedId=175632&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-175632">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Happens in 1.506 as well.
Also reported by <a href="http://stackoverflow.com/questions/15268736/error-publisher-org-jenkinsci-plugins-discardbuild-discardbuildpublisher-aborte" class="external-link" target="_blank" rel="nofollow noopener">http://stackoverflow.com/questions/15268736/error-publisher-org-jenkinsci-plugins-discardbuild-discardbuildpublisher-aborte</a></p>
</pre>
</details>

</ul>
</details>

<p>Happens in 1.506 as well.<br/>
Also reported by <a href="http://stackoverflow.com/questions/15268736/error-publisher-org-jenkinsci-plugins-discardbuild-discardbuildpublisher-aborte" class="external-link" target="_blank" rel="nofollow noopener">http://stackoverflow.com/questions/15268736/error-publisher-org-jenkinsci-plugins-discardbuild-discardbuildpublisher-aborte</a></p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>mwebber:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-16979?focusedId=175644&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-175644">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>It's not broken in 1.506 for me.</p>

<p>Note that the basic capability to discard old builds is part of Jenkins core, and that was broken in 1.503 and fixed in 1.504. This Jira ticket is about that.</p>

<p>In addition to the core functionality, there is a "Discard Old Build" plugin. DO you have that installed? Is that where the problem is? If so, you need to open a new ticket.</p>
</pre>
</details>

</ul>
</details>

<p>It's not broken in 1.506 for me.</p>

<p>Note that the basic capability to discard old builds is part of Jenkins core, and that was broken in 1.503 and fixed in 1.504. This Jira ticket is about that.</p>

<p>In addition to the core functionality, there is a "Discard Old Build" plugin. DO you have that installed? Is that where the problem is? If so, you need to open a new ticket.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/jglick.png" title="jglick's avatar" /> jglick:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-16979?focusedId=175676&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-175676">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>@bedge: right, file a ticket for that plugin if you need to.</p>
</pre>
</details>

</ul>
</details>

<p>@​bedge: right, file a ticket for that plugin if you need to.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/bedge.jpg" title="bedge's avatar" /> bedge:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-16979?focusedId=176867&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-176867">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Done, thanks.
<a href="https://issues.jenkins-ci.org/browse/JENKINS-17528" class="external-link" target="_blank" rel="nofollow noopener">https://issues.jenkins-ci.org/browse/JENKINS-17528</a></p>
</pre>
</details>

</ul>
</details>

<p>Done, thanks.<br/>
<a href="https://issues.jenkins-ci.org/browse/JENKINS-17528" class="external-link" target="_blank" rel="nofollow noopener">https://issues.jenkins-ci.org/browse/JENKINS-17528</a></p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/scm_issue_link.png" title="scm_issue_link's avatar" /> scm_issue_link:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-16979?focusedId=241577&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-241577">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Code changed in jenkins
User: Jesse Glick
Path:
 test/src/test/groovy/jenkins/model/BuildDiscarderTest.groovy
 test/src/test/java/jenkins/model/BuildDiscarderPropertyTest.java
 test/src/test/resources/jenkins/model/BuildDiscarderPropertyTest/logRotatorField/config.xml
 test/src/test/resources/jenkins/model/BuildDiscarderPropertyTest/logRotatorField/jobs/foo/config.xml
 test/src/test/resources/jenkins/model/BuildDiscarderTest/compatibility/config.xml
 test/src/test/resources/jenkins/model/BuildDiscarderTest/compatibility/jobs/foo/config.xml
<a href="http://jenkins-ci.org/commit/jenkins/2200711bf3dcd96263b3a88001a49f5b7241c264" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins-ci.org/commit/jenkins/2200711bf3dcd96263b3a88001a49f5b7241c264</a>
Log:
  <a href="https://issues.jenkins.io/browse/JENKINS-16979" title="&quot;Discard old builds&quot; broken in 1.503" class="issue-link" data-issue-key="JENKINS-16979"><del>JENKINS-16979</del></a> Migrated test from Groovy to Java for ease of maintenance.</p>


</pre>
</details>

</ul>
</details>

<p>Code changed in jenkins<br/>
User: Jesse Glick<br/>
Path:<br/>
 test/src/test/groovy/jenkins/model/BuildDiscarderTest.groovy<br/>
 test/src/test/java/jenkins/model/BuildDiscarderPropertyTest.java<br/>
 test/src/test/resources/jenkins/model/BuildDiscarderPropertyTest/logRotatorField/config.xml<br/>
 test/src/test/resources/jenkins/model/BuildDiscarderPropertyTest/logRotatorField/jobs/foo/config.xml<br/>
 test/src/test/resources/jenkins/model/BuildDiscarderTest/compatibility/config.xml<br/>
 test/src/test/resources/jenkins/model/BuildDiscarderTest/compatibility/jobs/foo/config.xml<br/>
<a href="http://jenkins-ci.org/commit/jenkins/2200711bf3dcd96263b3a88001a49f5b7241c264" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins-ci.org/commit/jenkins/2200711bf3dcd96263b3a88001a49f5b7241c264</a><br/>
Log:<br/>
  <a href="https://issues.jenkins.io/browse/JENKINS-16979" title=""Discard old builds" broken in 1.503" class="issue-link" data-issue-key="JENKINS-16979"><del>JENKINS-16979</del></a> Migrated test from Groovy to Java for ease of maintenance.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/dogfood.png" title="dogfood's avatar" /> dogfood:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-16979?focusedId=241583&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-241583">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Integrated in <span class="image-wrap" style=""><img src="https://ci.jenkins-ci.org/images/16x16/yellow.png" style="border: 0px solid black" /></span> <a href="https://ci.jenkins-ci.org/job/jenkins_main_trunk/4367/" class="external-link" target="_blank" rel="nofollow noopener">jenkins_main_trunk #4367</a>
     <a href="https://issues.jenkins.io/browse/JENKINS-16979" title="&quot;Discard old builds&quot; broken in 1.503" class="issue-link" data-issue-key="JENKINS-16979"><del>JENKINS-16979</del></a> Migrated test from Groovy to Java for ease of maintenance. (Revision 2200711bf3dcd96263b3a88001a49f5b7241c264)</p>

<p>     Result = UNSTABLE
jesse glick : <a href="https://github.com/jenkinsci/jenkins/commit/2200711bf3dcd96263b3a88001a49f5b7241c264" class="external-link" target="_blank" rel="nofollow noopener">2200711bf3dcd96263b3a88001a49f5b7241c264</a>
Files : </p>
<ul>
	<li>test/src/test/java/jenkins/model/BuildDiscarderPropertyTest.java</li>
	<li>test/src/test/resources/jenkins/model/BuildDiscarderPropertyTest/logRotatorField/jobs/foo/config.xml</li>
	<li>test/src/test/resources/jenkins/model/BuildDiscarderTest/compatibility/config.xml</li>
	<li>test/src/test/resources/jenkins/model/BuildDiscarderPropertyTest/logRotatorField/config.xml</li>
	<li>test/src/test/resources/jenkins/model/BuildDiscarderTest/compatibility/jobs/foo/config.xml</li>
	<li>test/src/test/groovy/jenkins/model/BuildDiscarderTest.groovy</li>
</ul>

</pre>
</details>

</ul>
</details>

<p>Integrated in <span class="image-wrap" style=""><img src="https://ci.jenkins-ci.org/images/16x16/yellow.png" style="border: 0px solid black" /></span> <a href="https://ci.jenkins-ci.org/job/jenkins_main_trunk/4367/" class="external-link" target="_blank" rel="nofollow noopener">jenkins_main_trunk #4367</a><br/>
     <a href="https://issues.jenkins.io/browse/JENKINS-16979" title=""Discard old builds" broken in 1.503" class="issue-link" data-issue-key="JENKINS-16979"><del>JENKINS-16979</del></a> Migrated test from Groovy to Java for ease of maintenance. (Revision 2200711bf3dcd96263b3a88001a49f5b7241c264)</p>

<p>     Result = UNSTABLE<br/>
jesse glick : <a href="https://github.com/jenkinsci/jenkins/commit/2200711bf3dcd96263b3a88001a49f5b7241c264" class="external-link" target="_blank" rel="nofollow noopener">2200711bf3dcd96263b3a88001a49f5b7241c264</a><br/>
Files : </p>
<ul>
	<li>test/src/test/java/jenkins/model/BuildDiscarderPropertyTest.java</li>
	<li>test/src/test/resources/jenkins/model/BuildDiscarderPropertyTest/logRotatorField/jobs/foo/config.xml</li>
	<li>test/src/test/resources/jenkins/model/BuildDiscarderTest/compatibility/config.xml</li>
	<li>test/src/test/resources/jenkins/model/BuildDiscarderPropertyTest/logRotatorField/config.xml</li>
	<li>test/src/test/resources/jenkins/model/BuildDiscarderTest/compatibility/jobs/foo/config.xml</li>
	<li>test/src/test/groovy/jenkins/model/BuildDiscarderTest.groovy</li>
</ul>

**Comment by jenkins-infra-bot** [comment]
<i>[Original `relates_to` from Jira: <a href="https://github.com/jenkinsci/jenkins/issues?q=is%3Aissue%20%22jira_issue_key%3DJENKINS-17528%22">JENKINS-17528</a>]</i>
